lib/src/components/avatars/avatar.dart

```diff
@@ -185,57 +202,204 @@ class ZetaAvatar extends StatelessWidget {
+  notification,
+}
+
+/// ZetaAvatarBadge component
+
+class ZetaAvatarBadge extends StatelessWidget {
+  /// Constructor for [ZetaAvatarBadge]
+  const ZetaAvatarBadge({
+    super.key,
+    this.color,
+    this.type = ZetaAvatarBadgeType.notification,
+    this.icon,
+    this.value,
+    this.iconColor,
+  }) : size = ZetaAvatarSize.xxxl;
+
+  const ZetaAvatarBadge._({
+    super.key,
+    required this.color,
+    required this.size,
+    required this.type,
+    this.icon,
+    this.value,
+    this.iconColor,
+  });
+
+  /// Constructs [ZetaAvatarBadge] with icon
+  const ZetaAvatarBadge.icon({
+    super.key,
+    this.color,
+    this.icon = ZetaIcons.star_round,
+    this.iconColor,
+  })  : value = null,
+        size = ZetaAvatarSize.xxxl,
+        type = ZetaAvatarBadgeType.icon;
+
+  /// Constructs [ZetaAvatarBadge] with notifications
+
+  const ZetaAvatarBadge.notification({
+    super.key,
+    required this.value,
+  })  : size = ZetaAvatarSize.xxxl,
+        icon = null,
+        iconColor = null,
+        color = null,
+        type = ZetaAvatarBadgeType.notification;
+
+  /// Size of badge
+  final ZetaAvatarSize size;
+
+  /// Type of badge
+  final ZetaAvatarBadgeType type;
+
+  /// Background color for badge
+  final Color? color;
+
+  /// Icon of badge
+  final IconData? icon;
+
+  /// Icon color for badge
+  final Color? iconColor;
+
+  /// Notification value for badge
+  final int? value;
+
+  /// Returns copy of [ZetaAvatarBadge]
+  ZetaAvatarBadge copyWith({
+    Color? color,
+    ZetaAvatarSize? size,
+    IconData? icon,
+    Color? iconColor,
+    int? value,
+    ZetaAvatarBadgeType? type,
+  }) {
+    return ZetaAvatarBadge._(
+      color: color ?? this.color,
+      size: size ?? this.size,
+      icon: icon ?? this.icon,
+      type: type ?? this.type,
+      value: value ?? this.value,
+      iconColor: iconColor ?? this.iconColor,
+      key: key,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = Zeta.of(context).colors;
+    final backgroundColor = type == ZetaAvatarBadgeType.notification ? colors.negative : color;
+    final badgeSize = _getContainerSize();
+    final borderSize = _getBorderSize();
+
+    final innerContent = Container(
+      margin: const EdgeInsets.all(0.01),
+      decoration: BoxDecoration(
+        color: backgroundColor,
+        borderRadius: ZetaRadius.full,
+      ),
+      child: value != null
+          ? Center(
+              child: Text(
+                value! > 99 ? '99+' : '$value',
+                style: TextStyle(
+                  color: backgroundColor?.onColor,
+                  fontSize: badgeSize / 2,
+                ),
+              ),
+            )
+          : icon != null
+              ? Icon(
+                  icon,
+                  size: badgeSize - borderSize,
+                  color: iconColor ?? backgroundColor?.onColor,
+                )
+              : null,
+    );
+
+    return Container(
```

Comment:
this should be pill-shaped if it is a notification:
![image](https://github.com/zebrafed/zeta_flutter/assets/23633655/1892bf49-0199-4187-8838-cd9e83877fff)
![image](https://github.com/zebrafed/zeta_flutter/assets/23633655/4ca67ddb-6e0e-4052-b836-866c75998940)